src/codegen/codegen_c_visitor.cpp

```diff
@@ -857,11 +857,21 @@ std::vector<SymbolType> CodegenCVisitor::get_float_variables() {
     if (info.vectorize) {
         variables.push_back(make_symbol(naming::VOLTAGE_UNUSED_VARIABLE));
     }
+
     if (breakpoint_exist()) {
         std::string name = info.vectorize ? naming::CONDUCTANCE_UNUSED_VARIABLE
                                           : naming::CONDUCTANCE_VARIABLE;
-        variables.push_back(make_symbol(name));
+
+        // make sure conductance variable like `g` is not already defined
+        auto iter =
+            std::find_if(variables.begin(), variables.end(), [&name](const SymbolType& sym) {
+                return name == sym->get_name();
+            });
+        if (iter == variables.end()) {
+            variables.push_back(make_symbol(name));
+        }
```

Comment:
C++17 !

```c++
if (auto r = std::find_if(variables.cbegin(), variables.cend(), [&](const SymbolType& s) { return name == s->get_name(); }); r == variables.cend()) {
    variables.push_back(make_symbol(name));
};
```

Reply:
We could maybe add range-v3 as a depends:

```c++
if (auto r = std::ranges::find_if(variables, [&](const SymbolType& s) { return name == s->get_name();}); r == variables.cend()) {
    variables.push_back(make_symbol(name));
}
```

Reply:
> C++17 !

thanks!